source/common/ur_singleton.hpp

```diff
@@ -45,7 +45,7 @@ template <typename singleton_tn, typename key_tn> class singleton_factory_t {
     /// the params are forwarded to the ctor of the singleton
     /// the first parameter must be the unique identifier of the instance
     template <typename... Ts> singleton_tn *getInstance(Ts &&...params) {
-        auto key = getKey(std::forward<Ts>(params)...);
+        auto key = getKey(params...);
```

Comment:
Why did you remove std::forward? It does not change anything with the current implementation but we might have some overloads in the future perhaps.